app/code/Magento/InventoryCatalog/Plugin/Model/ResourceModel/StockItemPlugin.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryCatalog\Plugin\Model\ResourceModel;
+
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\CatalogInventory\Model\ResourceModel\Stock\Item\Interceptor;
+use Magento\CatalogInventory\Model\Stock\Item;
+use Magento\Framework\Model\ResourceModel\Db\AbstractDb;
+use Magento\InventoryApi\Api\Data\SourceItemInterface;
+use Magento\InventoryApi\Api\SourceItemsSaveInterface;
+use Magento\InventoryCatalog\Model\DefaultSourceProvider;
+use Magento\Framework\Exception\AlreadyExistsException;
+
+/**
+ * Class provide Around Plugin on Stock::save to migrate single stock data
+ */
+class StockItemPlugin
```

Comment:
Provide more clear name (why is do in this plugin)

Reply:
yes, better to sync up with existing plugin we have for Reservation placement.
We could give them similar names and more to the single location.
For example,
```
1. UpdateLegacyCatalogInventoryAtStockDeductionTime
2. UpdateLegacyCatalogInventoryAtStockSettingTime
```
Because we have two different business operations with stock:
1. We set stock data and overwrite existing data in StockItem (despite the data stored there before) , this happens for example during the Import or synchronization with ERP
2. Stock Deduction, which happens during the order placement, when we need to subtract given quantity from the data we have in Stock